### DISCUSSION TITLE
There is a problem with calling Validation via a variable

### DISCUSSION THREAD

**Discussion by salawatbro** [dedoc/scramble#233] [Q&A] [answered]
That's how I wanted to use it:
$request->validate($data->rules());

<img width="773" alt="image" src="https://github.com/dedoc/scramble/assets/70514229/e0f9ef6d-98d7-4d2e-b261-79e2db63c59e">

But how can you fix such an error?
⚠️Cannot generate request documentation: Undefined variable $data

<img width="688" alt="image" src="https://github.com/dedoc/scramble/assets/70514229/f5128121-fd22-4942-afc9-fb7ee30256d6">

**Comment by CristianHG2** [comment]
Hello, I saw your message about this in the Scramble Discord. Scramble analyzes the code statically and is, therefore, unable to get the context of what happens during the runtime of your code. Evaluation of your variable includes that.

It's not necessarily impossible to provide native support for what you intend to do, but I believe it would require significant work to accomplish it elegantly.

Suppose this is a typical pattern throughout your code. In that case, I recommend using the Scramble Extension API to write an `OperationExtension` that can detect the service being used and grab the rules from it. You could even instantiate it via dependency injection as an argument of your controller method and detect it that way.

https://scramble.dedoc.co/developers/extensions

**Reply by WildEgo** [reply]
Hey I'm having this same exact issue, is there anyway to disable the static analysis for a specific `FormRequest`? The validation is body validation is created in real time depending on the request so I would rather have it be empty or just have some sort of generic

**Reply by CristianHG2** [reply]
You must write an extension that finds and runs the necessary code during analysis. I have a similar case with the `laravel-data` library, which generates validation rules on runtime. You can find this example here:

https://github.com/CristianHG2/laravel-data-scramble

**Reply by CristianHG2** [reply]
If you can send me a sample or screenshot of your code, I might be able to make some more accurate suggestions.

**Reply by WildEgo** [reply]
Hey mate I came up with this since in my case I just want a true generic, only issue being I get the annoying "warning" message as the description but I can live with it for now
```php
<?php

namespace App\Extensions;

use Dedoc\Scramble\Extensions\OperationExtension;
use Dedoc\Scramble\Support\Generator\Operation;
use Dedoc\Scramble\Support\Generator\RequestBodyObject;
use Dedoc\Scramble\Support\Generator\Schema;
use Dedoc\Scramble\Support\Generator\Types\ObjectType;
use Dedoc\Scramble\Support\Generator\Types\Type;
use Dedoc\Scramble\Support\RouteInfo;

class AnyType extends Type
{
    public function __construct()
    {
        parent::__construct('{}');
    }
}

class AddGenericBodyExtension extends OperationExtension
{
    public function handle(Operation $operation, RouteInfo $routeInfo)
    {
        if (\count($routeInfo->phpDoc()->getTagsByName('@genericBody'))) {
            $object = new ObjectType();

            $object->additionalProperties(new AnyType);

            $mediaType = 'multipart/form-data';
            if ($mediaTags = $routeInfo->phpDoc()->getTagsByName('@mediaType')) {
                $mediaType = trim(Arr::first($mediaTags)?->value?->value);
            }

            $requestBody = RequestBodyObject::make()->setContent(
                $mediaType,
                Schema::fromType($object)
            );

            $operation->addRequestBodyObject($requestBody);
        }
    }
}
```

**Comment by romalytvynenko** [comment, marked as answer]
Hey @SalawatJoldasbaev 

Can you try it like this?

```php
$request->validate(($data = app(ActivityStore::class))->rules());
```

The trick here is making sure to pass rules in a way so `eval` will work for them: https://scramble.dedoc.co/usage/request#rules-evaluation-caveats

**Reply by salawatbro** [reply]
Thank you bro. It worked!
